### PR TITLE
fixed: OC: when opening form and automatically going to dn widget on an image-map question, the image map ends up in the dn modal(add delay)

### PR DESCRIPTION
**Pull request by theywa**
#378

### CODE REVIEW

**Comment by MartijnR** [comment]
Hi Dewa, this was tested with the version of enketo-core that included your patch, right? Just asking, because we haven't deployed that patch to enketo-express-oc yet.

**Comment by theywa** [comment]
@MartijnR ~ yes

**Comment by MartijnR** [comment]
Thanks. Let's hold onto this until we merge enketo/enketo-express into enketo-express-oc, at which point you we do a final check that everything works.

File: widget/discrepancy-note/dn-widget.js [hunk 1]
```diff
@@ -561,11 +561,14 @@ class Comment extends Widget {
         } );
 
         // Scroll directly to Query modal if not enough height for user to view the Query modal
-        const queryModal = this.linkedQuestion.querySelector( '.widget.or-comment-widget' );
-        if ( window.innerHeight - this.linkedQuestion.offsetHeight < queryModal.offsetHeight ) {
-            queryModal.scrollIntoView( { behavior: 'smooth', block: 'start', inline: 'nearest' } );
+        // https://github.com/OpenClinica/enketo-express-oc/issues/378
+        if ( this.linkedQuestion.classList.contains( 'or-appearance-image-map' ) ) {
+            const that = this;
```
Comment (MartijnR) [inline widget/discrepancy-note/dn-widget.js:566]: little nitpicky, but would be nice to not use `that` and change timeout to an arrow function

File: widget/discrepancy-note/dn-widget.js [hunk 2]
```diff
@@ -580,6 +583,15 @@ class Comment extends Widget {
 
     }
 
+    _scrollToview() {
+        const queryModal = this.linkedQuestion.querySelector( '.widget.or-comment-widget' );
+        if ( window.innerHeight - this.linkedQuestion.offsetHeight < queryModal.offsetHeight ) {
+            queryModal.scrollIntoView( { behavior: 'smooth', block: 'start', inline: 'nearest' } );
```
Comment (MartijnR) [inline widget/discrepancy-note/dn-widget.js:589]: Also nitpicky, but I would put the options for scrollIntoView in a `const` since they're used twice.

**Review by MartijnR — APPROVED** [review]
Hi @theywa, 

If you agree, please change these 2 little things. 

I've added your enketo-core fix to the master branch of this repo, so maybe double-check that everything is working without manual edits in `node_modules`.

**Comment by theywa** [comment]
@MartijnR ~ sure no problem, code updated also I did check(including the new changes) and everything working as expected.

**Pull request merged**